apps/avifdec.c

```diff
@@ -248,17 +247,19 @@ int main(int argc, char * argv[])
                        decoder->image->height);
                 ++currIndex;
             }
-            if (nextImageResult != AVIF_RESULT_NO_IMAGES_REMAINING) {
-                printf("ERROR: Failed to decode frame: %s\n", avifResultToString(nextImageResult));
+            if (result != AVIF_RESULT_NO_IMAGES_REMAINING) {
```

Comment:
I suggest swapping the condition, I think it is easier to read:

```cpp
            if (result == AVIF_RESULT_NO_IMAGES_REMAINING) {
                result = AVIF_RESULT_OK;
            } else {
                assert(result != AVIF_RESULT_OK);
                printf("ERROR: Failed to decode frame: %s\n", avifResultToString(result));
                avifDumpDiagnostics(&decoder->diag);
            }
```

Reply:
Done.